index.js

```diff
@@ -171,27 +172,27 @@ module.exports = function (archive, target, opts, cb) {
     if (dryRun || entry) {
       next()
     } else {
-      next()
-      // TODO: not doing this in v8?
-      // archive.append({
-      //   name: hyperPath,
-      //   type: 'directory'
-      // }, next)
+      archive.mkdir(hyperPath, next)
     }
   }
 
   function next () {
-    consume(target, null, cb)
+    consume(target, null, done)
   }
 
   if (opts.resume) {
-    archive.history({ live: false })
-    .on('error', cb)
-    .on('data', function (entry) {
+    archive.history()
```

Comment:
Do we need to go through the whole history? AFAIK we only want to check against the latest file, and can use `archive.stat` instead.